features/document-write.yml

```
@@ -0,0 +1,15 @@
+name: document.write()
+description: The `document.open()`, `document.write()`, `document.writeln()`, and `document.close()` methods write content to a document, possibly while that document is still loading.
+spec: https://html.spec.whatwg.org/multipage/dynamic-markup-insertion.html
+discouraged:
+  # The spec doesn't actually discourage against open and close, but these methods are
+  # really only useful for writing anyway, so we're marking the whole feature as discouraged.
+  according_to:
+    - https://html.spec.whatwg.org/multipage/dynamic-markup-insertion.html#document.write()
+  alternatives:
+    - dom
+compat_features:
+  - api.Document.write
+  - api.Document.writeln
```

Oh interesting. I ran into `writeln()` independently because of trusted types stuff. BCD doesn't mark it as discouraged, but MDN does. I sent https://github.com/mdn/content/pull/38267 to fix MDN. It's one of those awkward cases where MDN got ahead of a discussion on the spec that never resolved (https://github.com/whatwg/html/issues/7977). I don't love it.

All that said, describing this overall _feature_ as discouraged seems correct to me though. This implies…

```suggestion
  - api.Document.write
  - api.Document.write.accepts_TrustedHTML
  - api.Document.writeln
  - api.Document.writeln.accepts_TrustedHTML
```